### ISSUE TITLE
$ref parameters not handled by the converter

### ISSUE THREAD

**Issue by turkeylurkey** [swagger-api/swagger-parser#455]
Swagger v2 parameters defined by $ref are modelled in v2 by the object RefParameter. When this type of parameter is encountered by the v3 converter the converter throws a class cast exception. 

On line 387:
```        
SerializableParameter sp = (SerializableParameter) v2Parameter;
```

This code works for the regular parameter types but RefParameter needs special handling in the converter.

**Comment by fehguy** [comment]
Can you please share an example?  Is this _only_ for v3 support?  Which file are you talking about?

**Comment by turkeylurkey** [comment]
The scenario is converting Swagger v2 to v3. I don't have a small, complete swagger file sample right now but I can create one. The style of parameters I'm referring to is:
```
        "parameters": [
          {
            "$ref": "#/parameters/serverID"
          },
          {
            "$ref": "#/parameters/additionalServerIDs"
          }
        ],
```
The converter file I'm referring to is line 387 of SwaggerConverter.java (io.swagger.parser.v2.SwaggerConverter).

**Comment by turkeylurkey** [comment]
Here is a complete sample:
```
{
  "swagger": "2.0",
  "info": {
    "title": "Uber API",
    "description": "Move your app forward with the Uber API",
    "version": "1.0.0"
  },
  "host": "api.uber.com",
  "schemes": [
    "https"
  ],
  "basePath": "/v1",
  "produces": [
    "application/json"
  ],
  "consumes" : [ "application/json" ],
  "paths": {
    "/api/batch/" : {
      "get" : {
        "tags" : [ "Batch" ],
        "summary" : "Get the list of resources",
        "description" : "Returns a list of available resources.",
        "schemes" : [ "https" ],
        "produces" : [ "application/json" ],
        "parameters" : [
          {
            "$ref" : "#/parameters/serverID"
          }
         ],
        "responses" : {
          "200" : {
            "description" : "successful operation",
          }
        }
      }
    },
  },
  "parameters": {
    "serverID": {
      "name": "serverID",
      "in": "path",
      "description": "serverID is: hostName,etc",
      "required": true,
      "type": "string"
    }
  },
  "definitions" : {
  }
}
```

**Comment by fehguy** [comment]
I looked at this, and added a [test](https://github.com/swagger-api/swagger-parser/pull/461/commits/dce12f18ed292ff7a34e1ac4c266112b4438605b).

The referenced parameters, in the 2.0 example, are correctly pulled in.  The behavior is the same as the 2.0 parser which--for better or worse--resolves the referenced parameters in-line.

Understanding that this isn't necessarily the goal of the 3.0 parser (that's why we created the ParseOptions), I also don't want to change the behavior of the 2.0 parser.  So unless I'm missing something, this can be closed.
